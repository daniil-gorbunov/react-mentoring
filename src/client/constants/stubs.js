export const testMovie = {
  id: 1,
  title: 'Test Movie',
  original_title: 'Test Movie',
  name: 'Test Movie',
  original_name: 'Test Movie',
  overview: 'Overview mocked',
  backdrop_path: 'backdrop.jpg',
  poster_path: 'poster.jpg',
  popularity: 100,
  vote_count: 100,
  vote_average: 5,
  genre_ids: [1],
  original_language: 'english',
  release_date: '2010-01-01',
  adult: false,
  video: true,
  origin_country: ['USA'],
  first_air_date: '2010-01-01',
};

export const testMovieWoPosters = {
  id: 2,
  title: 'Test Movie',
  original_title: 'Test Movie',
  name: 'Test Movie',
  original_name: 'Test Movie',
  overview: 'Overview mocked',
  backdrop_path: null,
  poster_path: null,
  popularity: 100,
  vote_count: 100,
  vote_average: 5,
  genre_ids: [1],
  original_language: 'english',
  release_date: '2010-01-01',
  adult: false,
  video: true,
  origin_country: ['USA'],
  first_air_date: '2010-01-01',
};

export const testMovieList = [
  testMovie,
  testMovieWoPosters,
];
