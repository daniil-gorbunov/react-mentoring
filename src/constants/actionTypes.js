export const CHANGE_SEARCH_QUERY = 'CHANGE_SEARCH_QUERY';
export const CHANGE_SEARCH_TYPE = 'CHANGE_SEARCH_TYPE';
export const CHANGE_SORT_TYPE = 'CHANGE_SORT_TYPE';
export const SORT_MOVIES = 'SORT_MOVIES';
export const SEARCH_MOVIES = 'SEARCH_MOVIES';
