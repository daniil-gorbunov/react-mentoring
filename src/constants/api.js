export const CONFIG = {
  IMAGES: {
    BASE_URL: 'http://image.tmdb.org/t/p/',
    POSTER_SIZES: {
      XS: 'w92',
      S: 'w154',
      M: 'w185',
      L: 'w342',
      XL: 'w500',
      XXL: 'w780',
      ORIGINAL: 'original',
    },
  },
};
export const BASE_URL = 'https://api.themoviedb.org/3/';
export const V3_KEY = '74613ce7f9c68dc71edc73f9216b3dd5';
