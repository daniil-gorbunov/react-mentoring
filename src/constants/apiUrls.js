export default 'https://netflixroulette.net/api/api.php';
