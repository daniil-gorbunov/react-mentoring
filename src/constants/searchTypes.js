export const TITLE = 'TITLE';
export const DIRECTOR = 'DIRECTOR';
export const SEARCH_TYPE_PARAMS_MAP = {
  [TITLE]: 'title',
  [DIRECTOR]: 'director',
};
