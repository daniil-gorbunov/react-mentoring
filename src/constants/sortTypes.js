export const DATE = 'DATE';
export const RATING = 'RATING';
export const SORT_TYPE_PARAMS_MAP = {
  [DATE]: 'release_date',
  [RATING]: 'vote_average',
};
