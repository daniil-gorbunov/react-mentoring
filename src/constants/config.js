import { CONFIG } from './api';

export const IMG_URL = CONFIG.IMAGES.BASE_URL;
export const POSTER_SIZE = CONFIG.IMAGES.POSTER_SIZES.L;
export const NO_IMAGE_PATH = '/src/assets/images/no-image.png';
