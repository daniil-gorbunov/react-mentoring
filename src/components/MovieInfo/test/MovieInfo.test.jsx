import React from 'react';
import renderer from 'react-test-renderer';
import { MovieInfoView } from '..';
import { testMovie } from '../../../constants/stubs';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('MovieInfo', () => {
  test('renders correctly', () => {
    const movieInfo = renderer.create(<MovieInfoView movie={testMovie} onSearchClick={() => {}} />);
    expect(movieInfo).toMatchSnapshot();
  });
});
