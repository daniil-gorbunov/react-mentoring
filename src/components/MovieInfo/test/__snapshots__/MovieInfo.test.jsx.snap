// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo renders correctly 1`] = `
<div
  className="movieInfo"
>
  <div
    className="row row between-xs"
  >
    <div
      className="col-xs-2"
    >
      <span
        className="title"
      >
        netflixroulette
      </span>
    </div>
    <div
      className="col-xs-2"
    >
      <div
        className="textRight"
      >
        <Link
          to="/"
        >
          <button
            className="searchBtn"
            onClick={[Function]}
          >
            Search
          </button>
        </Link>
      </div>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-4"
    >
      <img
        alt="Test Movie"
        className="poster"
        src="http://image.tmdb.org/t/p/w342poster.jpg"
      />
    </div>
    <div
      className="col-xs-8"
    >
      <div>
        <span
          className="movieTitle"
        >
          Test Movie
        </span>
        <div
          className="rating"
        >
          5
        </div>
      </div>
      <div>
        <span
          className="category"
        >
          100
           votes
        </span>
      </div>
      <div
        className="subHeader"
      >
        <span
          className="year"
        >
          2010
        </span>
      </div>
      <div>
        <p>
          Overview mocked
        </p>
      </div>
      <div
        className="cast"
      >
        <p>
          Popularity: 
          100
        </p>
        <p>
          Language: 
          english
        </p>
      </div>
    </div>
  </div>
</div>
`;
